x-pack/elastic-agent/pkg/agent/operation/operator.go

```diff
@@ -175,6 +179,7 @@ func (o *Operator) start(p Descriptor, cfg map[string]interface{}) (err error) {
 func (o *Operator) stop(p Descriptor) (err error) {
 	flow := []operation{
 		newOperationStop(o.logger, o.config),
+		newOperationUninstall(o.logger, p, o.uninstaller),
```

Comment:
what in case of multiple outputs running, you have metricbeat A and B A is stopped and uninstalled? it is the same binary as metricbeat for output B

Reply:
Is that even the case today? When looping with `detectPrograms` it only loops through the Supported programs once, creating an array. So only 1 metricbeat will be running.

The only case where that is not true with the sidecars for metrics and logs of the running metricbeat and filebeat.

At the moment uninstall doesn't do anything, other than call the `pre_uninstall` hooks. So this will not cause an issues currently.